4.analyze_data/notebooks/UMAP/2.find_sc.py

```diff
@@ -0,0 +1,198 @@
+
+    # Create 5 different file paths with "d0" through "d4" suffixes
+    for i in range(5):
+        filename = f"{images_dir}/{plate}_{well}{site}d{i}_illumcorrect.tiff"
+        file_paths.append(filename)
+
+    print("Randomly selected row:")
+    print(random_row)
+    print("Generated filenames:")
+    for path in file_paths:
+        print(path)
+
+    # Initialize empty lists to store the images for each channel
+    blue_channel = []
+    green_channel = []
+    red_channel = []
+
+    # Iterate through channels from the random well/site and assign the correct file names with the color channel
+    for file_path in file_paths:
+        filename = pathlib.Path(file_path).name
+        if 'd0' in filename:
+            blue_channel_image = cv2.imread(str(file_path), cv2.IMREAD_UNCHANGED)
+            blue_channel.append(blue_channel_image)
+        elif 'd1' in filename:
+            green_channel_image = cv2.imread(str(file_path), cv2.IMREAD_UNCHANGED)
+            green_channel.append(green_channel_image)
+        elif 'd4' in filename:
+            red_channel_image = cv2.imread(str(file_path), cv2.IMREAD_UNCHANGED)
+            red_channel.append(red_channel_image)
+
+    # Stack the images for each channel along the channel axis
+    blue_channel_stack = np.stack(blue_channel, axis=-1)
+    green_channel_stack = np.stack(green_channel, axis=-1)
+    red_channel_stack = np.stack(red_channel, axis=-1)
+
+    # Scale the pixel values to fit within the 16-bit range (0-65535)
+    blue_channel_stack = (blue_channel_stack / np.max(blue_channel_stack) * 65535).astype(np.uint16)
+    green_channel_stack = (green_channel_stack / np.max(green_channel_stack) * 65535).astype(np.uint16)
+    red_channel_stack = (red_channel_stack / np.max(red_channel_stack) * 65535).astype(np.uint16)
+
+    # Create the RGB numpy array by merging the channels
+    composite_image = cv2.merge((red_channel_stack, green_channel_stack, blue_channel_stack)).astype(np.uint16)
+
+    # Path for saving comp images
+    comp_path = pathlib.Path(f"{comp_dir}/{info['composite_save_path']}")
+
+    # Save the composite 16-bit RGB tiff image
+    tf.imwrite(comp_path, composite_image)
+
+    # Load the composite image from the save path as an Image object instead of numpy array
+    composite_image = Image.open(comp_path)
+
+    # Assuming you have a DataFrame called "filtered_df" with center coordinates
+    center_x = random_row["Metadata_Cells_Location_Center_X"]
+    center_y = random_row["Metadata_Cells_Location_Center_Y"]
+
+    # Define the size of the cropping box (250x250 pixels)
+    box_size = 250
+
+    # Paths for saving cropped images
+    crop_path = pathlib.Path(f"{crop_dir}/{info['crop_save_path']}")
+
+    # Iterate through the center coordinates and crop each cell
+    for x, y in zip(center_x, center_y):
+        left = x - box_size // 2
+        top = y - box_size // 2
+        right = x + box_size // 2
+        bottom = y + box_size // 2
```

Comment:
Double checking: floor division (`//`) is accurate to the intent of these lines, correct?

Reply:
I believe that floor division works best here as it rounds the integer to give exact bounding box coordinates each cropped single-cell. It might not make too much of a difference either way, so I will keep this method.